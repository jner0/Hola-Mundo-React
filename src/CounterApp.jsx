import PropTypes from 'prop-types'

const handleAdd = () => {
   console.log('+1') 
} 

export const CounterApp = ( {value} ) => {

    return(
      <>
        <h1> Counter App </h1>
        <h2> { value } </h2>

        <button onClick={ handleAdd }>
          +1
        </button>
      </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

