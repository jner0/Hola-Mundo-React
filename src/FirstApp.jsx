//import { Fragment } from "react";
import PropTypes from 'prop-types';

const getResult = (a,b) =>{
  return a + b;
};

export const FirstApp = ( {title, subTitle, name} ) => {
  
  return (       
    //<Fragment>
    <>  
      <h1>{ title }</h1>
      {/* <code> { JSON.stringify(newMessage) } </code> */}
      <p>{subTitle }</p>
      <p>{name}</p>
    </>  
    //</Fragment>

  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: 'No hay nungun Titulo',
  subTitle: 'No Hay Subtitulo',
  name: 'Nemecio Rodriguez',
}