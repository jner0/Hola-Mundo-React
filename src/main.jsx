import React from 'react';
import ReactDom from 'react-dom/client';

import {HelloWorldApp}  from './HelloWorldApp';
import {FirstApp} from './FirstApp'
import { CounterApp } from './CounterApp';

import './style.css'

ReactDom.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={20}></CounterApp>        
        {/* <FirstApp title='Hola, soy Vegeta'/> */}
    </React.StrictMode> 
)