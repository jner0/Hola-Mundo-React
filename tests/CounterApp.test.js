import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Prubas en CounterApp', () => { 

    const initialValue = 10;

    test('Debe hacer match con el snapshpt', () => { 

        const { container } =render( <CounterApp value={ initialValue }/> )

        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el valor inicial de 100 counterApp', () => { 

        render( <CounterApp value={ 100 } /> );
        expect( screen.getByText('100') ).toBeTruthy();
    });

    test('Debe de incrementar el boton +1', () => { 

        render( <CounterApp value={ initialValue }/> );
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('11') ).toBeTruthy();

    });

    test('Debe de decrementar el boton -1', () => { 

        render( <CounterApp value={ initialValue }/> );
        fireEvent.click( screen.getByText('-1') );
        //screen.debug();
        expect( screen.getByText('9') ).toBeTruthy();

    });


    test('Debe de funcionar el boton de reset', () => { 

        render( <CounterApp value={ initialValue }/> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        //fireEvent.click( screen.getByText('Reset') );
        fireEvent.click (screen.getByRole( 'button', { name: 'btn-reset' } ));

        //expect( screen.getByText( initialValue ) ).toBeTruthy();

        //screen.debug();

        

    });


});