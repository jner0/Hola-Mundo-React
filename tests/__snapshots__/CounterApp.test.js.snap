// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en CounterApp Debe hacer match con el snapshpt 1`] = `
<div>
  <h1>
     Contador 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
